### ISSUE TITLE
[BUG] VIP is temporarily not accessible when upgrading

### ISSUE THREAD

**Issue by Vicente-Cheng** [harvester/harvester#4221]
**Describe the bug**
VIP is temporarily not accessible when upgrading.
The VIP will be unreachable when we upgrade with stage `apply manifest`.
That will cause some services which depend on VIP to go down.

**To Reproduce**
1. create a v1.1.2 cluster
2. upgrade to v1.2.0-rc3
3. Check the VIP connection, it should be down with stage `apply manifest`.

**Expected behavior**
VIP should not be down for more than 30 seconds.

**Support bundle**

**Environment**
 - Harvester ISO version: v1.2.0-rc3
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): 3 VM cluster

**Additional context**

**Comment by bk201** [comment]
I can reproduce this issue with these steps:
- Create a single node v1.1.2 cluster.
- Run [this script](https://raw.githubusercontent.com/bk201/misc/main/harvester/issue-4221/upgrade-manifest.sh) to bump the Harvester chart to v1.2.0-rc3. 

VIP is gone and back, but I can't ping it (even in the node):
```
4: mgmt-br: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default q
len 1000
    link/ether 52:54:00:6a:59:7c brd ff:ff:ff:ff:ff:ff
    inet 10.10.0.11/24 brd 10.10.0.255 scope global mgmt-br
       valid_lft forever preferred_lft forever
    inet 10.10.0.100/32 scope global mgmt-br
       valid_lft forever preferred_lft forever
```

```
4: mgmt-br: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default q
len 1000
    link/ether 52:54:00:6a:59:7c brd ff:ff:ff:ff:ff:ff
    inet 10.10.0.11/24 brd 10.10.0.255 scope global mgmt-br
       valid_lft forever preferred_lft forever
```

```
4: mgmt-br: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default q
len 1000
    link/ether 52:54:00:6a:59:7c brd ff:ff:ff:ff:ff:ff
    inet 10.10.0.11/24 brd 10.10.0.255 scope global mgmt-br
       valid_lft forever preferred_lft forever
    inet 10.10.0.100/32 scope global mgmt-br
       valid_lft forever preferred_lft forever
```

**Comment by yaocw2020** [comment]
Reason: The Harvester pod adds the annotation `kube-vip.io/ignore-service-security: "true"` to the service ingress-expose, but the kube-vip can't handle the modification.

Workaround: Delete kube-vip pod.

**Comment by TachunLin** [comment]
### Test Plans

#### Single nodes upgrade 

- [x] Check can reach the VIP during the upgrading Harvester

    <details><summary>Test Result $~~$ $\color{green}{\textsf{PASS}}$ </summary>  
    
     * During the upgrade process in `upgrading service`,  we encounter a short time (less than 30 seconds) unable to access the VIP
        ![image](https://github.com/harvester/harvester/assets/29251855/80620cae-d183-41fe-a3c1-c5e8b007f9d2)
        ![image](https://github.com/harvester/harvester/assets/29251855/8cab0c11-fbcf-4811-988c-58076bf203bf)
        ![image](https://github.com/harvester/harvester/assets/29251855/24083154-334d-4ec9-b033-a05ecd0d2163)
    
     * In `upgrading node` stage, the VIP is also not accessible, this would be expected since node is rebooting 
      ![image](https://github.com/harvester/harvester/assets/29251855/0eee9d08-8c00-4587-b488-500d7ba06e66)
    
     * And we can complete the upgrade 
      ![image](https://github.com/harvester/harvester/assets/29251855/ed574bb8-1ba6-4c92-864e-05ebbb62409c)
    
    </details> 


#### Multi nodes upgrade 

- [x] Check can reach the VIP during the upgrading Harvester

    <details><summary>Test Result $~~$ $\color{green}{\textsf{PASS}}$ </summary><br>

     * During the entire upgrade process in `upgrading service`,  we did not encounter missing the VIP connection
      
     ![image](https://github.com/harvester/harvester/assets/29251855/20eda1aa-d773-440d-b73a-a51d4b18e220)
     ![image](https://github.com/harvester/security/assets/29251855/2f107787-4c0f-4dab-a7aa-b0e3cc4fcf73)


     * In `upgrading node` stage, we did not lost the VIP connection too.
     ![image](https://github.com/harvester/harvester/assets/29251855/609e2292-bccb-4d03-8f15-910b439d1005)
    
     * And we can complete the upgrade 
      ![image](https://github.com/harvester/harvester/assets/29251855/ed574bb8-1ba6-4c92-864e-05ebbb62409c)
    

  
    </details>

**Comment by TachunLin** [comment]
Verified **fixed** on Harvester `v1.2-57b26312-head` (08/14). Close this issue.

### Result
We passed all the test scenarios listed in https://github.com/harvester/harvester/issues/4221#issuecomment-1680103629

The VIP accessibility did not lost during the `upgrade service` stage (apply manifest job) in:

  
- [x] 1. On single node we can reach the VIP during the upgrading Harvester, 
   - The VIP only stop within 60 seconds then come back, this would be expected since it's single node and restart the kube-vip

- [x] 2. On multiple nodes we can always reach the VIP during the upgrading Harvester and the follow up upgrade nodes stage. 


### Test Information
* Test Environment: Three nodes on bare machines and single node on local kvm 
* Harvester version: `v1.2-57b26312-head` (08/14)

### Verify Steps

<details><summary> Single node steps </summary><br>

  1. Prepare a single node v1.1.2 Harvester
  2. Upgrade to v1.2-57b26312-head 
  3. ssh to management node, and run k9s
  4. Search jobs 
  5. Open terminal and ping the VIP  
  6. Check can ping VIP during the entire upgrade service process 
  7. When the apply manifest job exists (After upgrade service started), check it's status
   
</details>

<details><summary> Multiple nodes steps </summary><br>

  1. Prepare a three node v1.1.2 Harvester
  2. Upgrade to v1.2-57b26312-head 
  3. ssh to management node, and run k9s
  4. Search jobs 
  5. Open terminal and ping the VIP  
  6. Check can ping VIP during the entire upgrade service process 
  7. When the apply manifest job exists (After upgrade service started), check it's status
  8. Check can ping VIP during the follow up upgrade node process

</details>
